### DISCUSSION TITLE
JSON $schema version updates

### DISCUSSION THREAD

**Discussion by setchy** [renovatebot/renovate#28535] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

We're using Biome as our formatter+linter.

It's configuration is defined in a JSON file, and contains a schema definition that matches the version of biome in our package.json

For example
**biome.json**
```json
{
  "$schema": "https://biomejs.dev/schemas/1.7.0/schema.json",
  ...
}
```

**package.json**
```json
{
  ...
  "dependencies": {
    "@biomejs/biome": "1.7.0"
  }
  ...
}
```

Seeking any examples or suggestions on how we can keep the version in sync across the package.json and $schema def

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
The custom datasource now supports local files, but that would mean you update the dependency first and then the schema update would be picked up only after merge. Instead, how about using a custom manager and pointing it to the npm package for versions? Make sure to give it the same depName and it will be upgraded together in the same PR

**Reply by setchy** [reply, marked as answer]
Thanks @rarkins - that worked quite nicely

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:recommended"
  ],
  "customManagers": [
    {
      "customType": "regex",
      "fileMatch": ["^biome.json$"],
      "matchStrings": ["https://biomejs.dev/schemas/(?<currentValue>.*)/schema.json"],
      "depNameTemplate": "@biomejs/biome",
      "datasourceTemplate": "npm"
    }
  ]
}
```

**Reply by setchy** [reply]
Contributed as a shared regex-manager preset: https://github.com/renovatebot/renovate/pull/28537
